server/server.go

```diff
@@ -132,7 +132,7 @@ var backoff = wait.Backoff{
 
 var (
 	clientConstraint = fmt.Sprintf(">= %s", common.MinClientVersion)
-	baseHRefRegex    = regexp.MustCompile(`<base href="(.*)">`)
+	baseHRefRegex    = regexp.MustCompile(`<base href="(.*?)">`)
```

Comment:
Can you please add a test as well? I would suggest moving the line below into function and add a unit test for it 

https://github.com/argoproj/argo-cd/blob/a67b97d6ad5d17137190f6d9b863757da97e7c82/server/server.go#L896

Reply:
added unit test